jobs: TestBatchJobsCreation failed

jobs.TestBatchJobsCreation [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3303308&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3303308&tab=artifacts#/) on master @ [ee3efd6b1e24a3e1676778f5028fa0a35266f683](https://github.com/cockroachdb/cockroach/commits/ee3efd6b1e24a3e1676778f5028fa0a35266f683):


```
E210815 05:21:05.561462 73617 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5447  making txn commit explicit failed for "unnamed" meta={id=f268ee34 pri=0.00281599 epo=0 ts=1629004842.613969774,0 min=1629004842.613969774,0 seq=4} lock=true stat=STAGING rts=1629004842.613969774,0 wto=false gul=1629004843.113969774,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.561589 96916 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5448  making txn commit explicit failed for "unnamed" meta={id=0ac5ec6d pri=0.00871411 epo=0 ts=1629004845.881803359,0 min=1629004845.881803359,0 seq=4} lock=true stat=STAGING rts=1629004845.881803359,0 wto=false gul=1629004846.381803359,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.561732 69523 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5449  making txn commit explicit failed for "unnamed" meta={id=3bf2d6f0 pri=0.05552266 epo=0 ts=1629004841.416280630,0 min=1629004841.416280630,0 seq=4} lock=true stat=STAGING rts=1629004841.416280630,0 wto=false gul=1629004841.916280630,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.561854 77888 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5450  making txn commit explicit failed for "unnamed" meta={id=59730df7 pri=0.07100096 epo=0 ts=1629004843.688616970,0 min=1629004843.688616970,0 seq=4} lock=true stat=STAGING rts=1629004843.688616970,0 wto=false gul=1629004844.188616970,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.561972 96508 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5451  making txn commit explicit failed for "unnamed" meta={id=364611b2 pri=0.03459323 epo=0 ts=1629004844.556538980,0 min=1629004844.556538980,0 seq=4} lock=true stat=STAGING rts=1629004844.556538980,0 wto=false gul=1629004845.056538980,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.562088 75139 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5452  making txn commit explicit failed for "unnamed" meta={id=b611c93e pri=0.01534647 epo=0 ts=1629004842.725549480,0 min=1629004842.725549480,0 seq=4} lock=true stat=STAGING rts=1629004842.725549480,0 wto=false gul=1629004843.225549480,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.562587 96562 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5453  making txn commit explicit failed for "unnamed" meta={id=703fd1c6 pri=0.00566067 epo=0 ts=1629004845.712358823,0 min=1629004845.712358823,0 seq=4} lock=true stat=STAGING rts=1629004845.712358823,0 wto=false gul=1629004846.212358823,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.562725 68827 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5454  making txn commit explicit failed for "unnamed" meta={id=e23d4587 pri=0.00225552 epo=0 ts=1629004841.228879885,0 min=1629004841.228879885,0 seq=8} lock=true stat=STAGING rts=1629004841.228879885,0 wto=false gul=1629004841.728879885,0 ifw=7: result is ambiguous (server shutdown)
E210815 05:21:05.563041 96501 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5455  making txn commit explicit failed for "unnamed" meta={id=52afd52e pri=0.03301711 epo=0 ts=1629004844.106326487,0 min=1629004844.106326487,0 seq=4} lock=true stat=STAGING rts=1629004844.106326487,0 wto=false gul=1629004844.606326487,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563056 74514 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5456  making txn commit explicit failed for "unnamed" meta={id=84a13d2d pri=0.02475115 epo=0 ts=1629004842.382919374,0 min=1629004842.382919374,0 seq=4} lock=true stat=STAGING rts=1629004842.382919374,0 wto=false gul=1629004842.882919374,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563237 96681 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5457  making txn commit explicit failed for "unnamed" meta={id=dcb77e37 pri=0.01350595 epo=0 ts=1629004845.935737039,0 min=1629004845.935737039,0 seq=4} lock=true stat=STAGING rts=1629004845.935737039,0 wto=false gul=1629004846.435737039,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563383 96430 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5458  making txn commit explicit failed for "unnamed" meta={id=3239bb32 pri=0.02050698 epo=0 ts=1629004845.221880888,0 min=1629004845.221880888,0 seq=4} lock=true stat=STAGING rts=1629004845.221880888,0 wto=false gul=1629004845.721880888,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563498 68674 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5459  making txn commit explicit failed for "unnamed" meta={id=05abc1e8 pri=0.02792072 epo=0 ts=1629004841.719269025,0 min=1629004841.719269025,0 seq=4} lock=true stat=STAGING rts=1629004841.719269025,0 wto=false gul=1629004842.219269025,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563605 92823 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5460  making txn commit explicit failed for "unnamed" meta={id=a24d32f9 pri=0.01115901 epo=0 ts=1629004843.480429843,0 min=1629004843.480429843,0 seq=4} lock=true stat=STAGING rts=1629004843.480429843,0 wto=false gul=1629004843.980429843,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563716 68683 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5461  making txn commit explicit failed for "unnamed" meta={id=902e1654 pri=0.01609852 epo=0 ts=1629004841.353098004,0 min=1629004841.353098004,0 seq=4} lock=true stat=STAGING rts=1629004841.353098004,0 wto=false gul=1629004841.853098004,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563831 75566 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5462  making txn commit explicit failed for "unnamed" meta={id=016f2045 pri=0.00690441 epo=0 ts=1629004843.078343379,0 min=1629004843.078343379,0 seq=4} lock=true stat=STAGING rts=1629004843.078343379,0 wto=false gul=1629004843.578343379,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.563942 96363 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5463  making txn commit explicit failed for "unnamed" meta={id=866db0cb pri=0.03284188 epo=0 ts=1629004845.924370025,0 min=1629004845.924370025,0 seq=4} lock=true stat=STAGING rts=1629004845.924370025,0 wto=false gul=1629004846.424370025,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564046 78353 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5464  making txn commit explicit failed for "unnamed" meta={id=81685dc4 pri=0.00245110 epo=0 ts=1629004843.803510326,0 min=1629004843.803510326,0 seq=4} lock=true stat=STAGING rts=1629004843.803510326,0 wto=false gul=1629004844.303510326,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564147 68206 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5465  making txn commit explicit failed for "unnamed" meta={id=e0f43808 pri=0.03254213 epo=0 ts=1629004841.392096278,0 min=1629004841.392096278,0 seq=4} lock=true stat=STAGING rts=1629004841.392096278,0 wto=false gul=1629004841.892096278,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564268 78010 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5466  making txn commit explicit failed for "unnamed" meta={id=e5123589 pri=0.10244120 epo=0 ts=1629004843.824550751,0 min=1629004843.824550751,0 seq=4} lock=true stat=STAGING rts=1629004843.824550751,0 wto=false gul=1629004844.324550751,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564385 78282 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5467  making txn commit explicit failed for "unnamed" meta={id=7687ac0b pri=0.02339277 epo=0 ts=1629004843.012071595,0 min=1629004843.012071595,0 seq=4} lock=true stat=STAGING rts=1629004843.012071595,0 wto=false gul=1629004843.512071595,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564493 96622 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5468  making txn commit explicit failed for "unnamed" meta={id=a6f8230f pri=0.03071246 epo=0 ts=1629004844.546483235,0 min=1629004844.546483235,0 seq=4} lock=true stat=STAGING rts=1629004844.546483235,0 wto=false gul=1629004845.046483235,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564617 68662 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5469  making txn commit explicit failed for "unnamed" meta={id=b15ea34b pri=0.00953726 epo=0 ts=1629004841.491060347,0 min=1629004841.491060347,0 seq=4} lock=true stat=STAGING rts=1629004841.491060347,0 wto=false gul=1629004841.991060347,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564728 69047 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5470  making txn commit explicit failed for "unnamed" meta={id=552087bb pri=0.01689000 epo=0 ts=1629004842.109884143,0 min=1629004842.109884143,0 seq=4} lock=true stat=STAGING rts=1629004842.109884143,0 wto=false gul=1629004842.609884143,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564835 69271 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5471  making txn commit explicit failed for "unnamed" meta={id=a2fe009c pri=0.01058709 epo=0 ts=1629004841.922620994,0 min=1629004841.922620994,0 seq=4} lock=true stat=STAGING rts=1629004841.922620994,0 wto=false gul=1629004842.422620994,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.564952 69407 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5472  making txn commit explicit failed for "unnamed" meta={id=51501bad pri=0.05363910 epo=0 ts=1629004842.149906333,0 min=1629004842.149906333,0 seq=4} lock=true stat=STAGING rts=1629004842.149906333,0 wto=false gul=1629004842.649906333,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565225 96407 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5473  making txn commit explicit failed for "unnamed" meta={id=656f3f7c pri=0.02059122 epo=0 ts=1629004845.390368929,0 min=1629004845.390368929,0 seq=4} lock=true stat=STAGING rts=1629004845.390368929,0 wto=false gul=1629004845.890368929,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565358 75313 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5474  making txn commit explicit failed for "unnamed" meta={id=3ae537eb pri=0.02419767 epo=0 ts=1629004842.251081045,0 min=1629004842.251081045,0 seq=4} lock=true stat=STAGING rts=1629004842.251081045,0 wto=false gul=1629004842.751081045,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565471 79529 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5475  making txn commit explicit failed for "unnamed" meta={id=88dadb04 pri=0.03818404 epo=0 ts=1629004842.311119915,0 min=1629004842.311119915,0 seq=4} lock=true stat=STAGING rts=1629004842.311119915,0 wto=false gul=1629004842.811119915,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565578 80615 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5476  making txn commit explicit failed for "unnamed" meta={id=f67da8cb pri=0.00353148 epo=0 ts=1629004844.007459533,0 min=1629004844.007459533,0 seq=4} lock=true stat=STAGING rts=1629004844.007459533,0 wto=false gul=1629004844.507459533,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565700 68478 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5477  making txn commit explicit failed for "unnamed" meta={id=0d5c00cd pri=0.03921059 epo=0 ts=1629004841.492986937,0 min=1629004841.492986937,0 seq=4} lock=true stat=STAGING rts=1629004841.492986937,0 wto=false gul=1629004841.992986937,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565812 74917 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5478  making txn commit explicit failed for "unnamed" meta={id=e58befa4 pri=0.00894899 epo=0 ts=1629004842.091694621,0 min=1629004842.091694621,0 seq=4} lock=true stat=STAGING rts=1629004842.091694621,0 wto=false gul=1629004842.591694621,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.565934 72537 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5479  making txn commit explicit failed for "unnamed" meta={id=77e4718a pri=0.00521569 epo=0 ts=1629004842.380205103,0 min=1629004842.380205103,0 seq=4} lock=true stat=STAGING rts=1629004842.380205103,0 wto=false gul=1629004842.880205103,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566052 79977 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5480  making txn commit explicit failed for "unnamed" meta={id=8bd3ea3a pri=0.01900191 epo=0 ts=1629004843.341981315,0 min=1629004843.341981315,0 seq=4} lock=true stat=STAGING rts=1629004843.341981315,0 wto=false gul=1629004843.841981315,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566170 71544 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5481  making txn commit explicit failed for "unnamed" meta={id=bc246967 pri=0.00704397 epo=0 ts=1629004842.259595951,0 min=1629004842.259595951,0 seq=4} lock=true stat=STAGING rts=1629004842.259595951,0 wto=false gul=1629004842.759595951,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566299 80085 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5482  making txn commit explicit failed for "unnamed" meta={id=cb76ca64 pri=0.00316766 epo=0 ts=1629004843.981813795,0 min=1629004843.981813795,0 seq=4} lock=true stat=STAGING rts=1629004843.981813795,0 wto=false gul=1629004844.481813795,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566408 72127 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5483  making txn commit explicit failed for "unnamed" meta={id=8f1ed073 pri=0.02334570 epo=0 ts=1629004841.757803819,0 min=1629004841.757803819,0 seq=4} lock=true stat=STAGING rts=1629004841.757803819,0 wto=false gul=1629004842.257803819,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566518 71546 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5484  making txn commit explicit failed for "unnamed" meta={id=b23d7630 pri=0.01703245 epo=0 ts=1629004841.836027142,0 min=1629004841.836027142,0 seq=4} lock=true stat=STAGING rts=1629004841.836027142,0 wto=false gul=1629004842.336027142,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566630 87276 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5485  making txn commit explicit failed for "unnamed" meta={id=0c229df8 pri=0.00163764 epo=0 ts=1629004844.355016698,0 min=1629004844.355016698,0 seq=4} lock=true stat=STAGING rts=1629004844.355016698,0 wto=false gul=1629004844.855016698,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.566743 72294 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5486  making txn commit explicit failed for "unnamed" meta={id=b6c2f9c6 pri=0.01731762 epo=0 ts=1629004842.323414826,0 min=1629004842.323414826,0 seq=4} lock=true stat=STAGING rts=1629004842.323414826,0 wto=false gul=1629004842.823414826,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.567295 71548 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5487  making txn commit explicit failed for "unnamed" meta={id=c875a55b pri=0.00921064 epo=0 ts=1629004842.073165552,0 min=1629004842.073165552,0 seq=4} lock=true stat=STAGING rts=1629004842.073165552,0 wto=false gul=1629004842.573165552,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.567485 96677 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5488  making txn commit explicit failed for "unnamed" meta={id=f15fa3cb pri=0.04006126 epo=0 ts=1629004846.034917463,0 min=1629004846.034917463,0 seq=4} lock=true stat=STAGING rts=1629004846.034917463,0 wto=false gul=1629004846.534917463,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.567676 86960 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5489  making txn commit explicit failed for "unnamed" meta={id=088d15db pri=0.01838841 epo=0 ts=1629004843.945330660,0 min=1629004843.945330660,0 seq=4} lock=true stat=STAGING rts=1629004843.945330660,0 wto=false gul=1629004844.445330660,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.567902 80230 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5490  making txn commit explicit failed for "unnamed" meta={id=8f512382 pri=0.02215561 epo=0 ts=1629004843.698317016,0 min=1629004843.698317016,0 seq=4} lock=true stat=STAGING rts=1629004843.698317016,0 wto=false gul=1629004844.198317016,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.568188 72214 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5491  making txn commit explicit failed for "unnamed" meta={id=a5391493 pri=0.07185219 epo=0 ts=1629004842.234967732,0 min=1629004842.234967732,0 seq=4} lock=true stat=STAGING rts=1629004842.234967732,0 wto=false gul=1629004842.734967732,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.568258 71550 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5492  making txn commit explicit failed for "unnamed" meta={id=d9303622 pri=0.00912384 epo=0 ts=1629004841.546769167,0 min=1629004841.546769167,0 seq=4} lock=true stat=STAGING rts=1629004841.546769167,0 wto=false gul=1629004842.046769167,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.568423 93199 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5493  making txn commit explicit failed for "unnamed" meta={id=32cd6538 pri=0.01563248 epo=0 ts=1629004843.955821716,0 min=1629004843.955821716,0 seq=4} lock=true stat=STAGING rts=1629004843.955821716,0 wto=false gul=1629004844.455821716,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.568574 71538 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5494  making txn commit explicit failed for "unnamed" meta={id=53bbac28 pri=0.03707954 epo=0 ts=1629004842.295937239,0 min=1629004842.295937239,0 seq=4} lock=true stat=STAGING rts=1629004842.295937239,0 wto=false gul=1629004842.795937239,0 ifw=3: result is ambiguous (server shutdown)
E210815 05:21:05.568722 71972 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 5495  making txn commit explicit failed for "unnamed" meta={id=f95be830 pri=0.00616028 epo=0 ts=1629004842.206984648,0 min=1629004842.206984648,0 seq=4} lock=true stat=STAGING rts=1629004842.206984648,0 wto=false gul=1629004842.706984648,0 ifw=3: result is ambiguous (server shutdown)
    --- FAIL: TestBatchJobsCreation/extra_large_batch (33.08s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBatchJobsCreation PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-schema sajjadrizvi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBatchJobsCreation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/registry.go
+++ b/pkg/jobs/registry.go
@@ -1313,6 +1313,9 @@ func (r *Registry) stepThroughStateMachine(
 }
 
 func (r *Registry) adoptionDisabled(ctx context.Context) bool {
+	if r.knobs.DisableAdoptions {
+		return true
+	}
 	if r.preventAdoptionFile != "" {
 		if _, err := os.Stat(r.preventAdoptionFile); err != nil {
 			if !oserror.IsNotExist(err) {

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -310,7 +310,10 @@ func TestBatchJobsCreation(t *testing.T) {
 
 				args := base.TestServerArgs{
 					Knobs: base.TestingKnobs{
-						JobsTestingKnobs: NewTestingKnobsWithShortIntervals(),
+						// Avoiding jobs to be adopted.
+						JobsTestingKnobs: &TestingKnobs{
+							DisableAdoptions: true,
+						},
 					},
 				}
 
@@ -345,13 +348,8 @@ func TestBatchJobsCreation(t *testing.T) {
 					return err
 				}))
 				require.Equal(t, len(jobIDs), test.batchSize)
-				// Wait for the jobs to complete.
-				tdb.CheckQueryResultsRetry(t, "SELECT count(*) FROM [SHOW JOBS]",
+				tdb.CheckQueryResults(t, "SELECT count(*) FROM [SHOW JOBS]",
 					[][]string{{fmt.Sprintf("%d", test.batchSize)}})
-				for _, id := range jobIDs {
-					tdb.CheckQueryResultsRetry(t, fmt.Sprintf("SELECT status FROM system.jobs WHERE id = '%d'", id),
-						[][]string{{"succeeded"}})
-				}
 			}
 		})
 	}

--- a/pkg/jobs/testing_knobs.go
+++ b/pkg/jobs/testing_knobs.go
@@ -63,6 +63,9 @@ type TestingKnobs struct {
 
 	// TimeSource replaces registry's clock.
 	TimeSource *hlc.Clock
+
+	// DisableAdoptions disables job adoptions.
+	DisableAdoptions bool
 }
 
 // ModuleTestingKnobs is part of the base.ModuleTestingKnobs interface.

